### DISCUSSION TITLE
Heuristic Prefetching

### DISCUSSION THREAD

**Discussion by VinayaSathyanarayana** [vercel/next.js#25185] [Ideas]
### Describe the feature you'd like to request

Prefetch pages when the user hovers on a link

### Describe the solution you'd like

Leverage [Guess.js](https://github.com/guess-js/guess) with a Next.js application. It provides heuristic pre-fetching.

### Describe alternatives you've considered

none

**Comment by LucaNerlich** [comment]
next/link does this ootb: https://nextjs.org/docs/api-reference/next/link

Or do you want to prefetch external hrefs? This should always be custom code / per project imo.
